test/compute_bands.jl

```diff
@@ -168,46 +162,27 @@ end
     ret = DFTK.prepare_band_data((basis=basis, λ=λ, λerror=λerror), klabels=klabels)
 
     @test ret.n_spin   == 1
-    @test ret.n_kcoord == 9
+    @test ret.n_kcoord == 8
     @test ret.n_bands  == 4
 
-    @test ret.branches[1].kindices == [1, 2, 3]
-    @test ret.branches[2].kindices == [4, 5, 6]
-    @test ret.branches[3].kindices == [7, 8, 9]
-
-    @test ret.branches[1].klabels == ("Γ", "X")
-    @test ret.branches[2].klabels == ("X", "W")
-    @test ret.branches[3].klabels == ("U", "X")
-
     for iband in 1:4
-        @test ret.branches[1].λ[:, iband, 1] == [10ik .+ iband for ik in 1:3]
-        @test ret.branches[2].λ[:, iband, 1] == [10ik .+ iband for ik in 4:6]
-        @test ret.branches[3].λ[:, iband, 1] == [10ik .+ iband for ik in 7:9]
-
-        for ibr in 1:3
-            @test ret.branches[ibr].λerror[:, iband, 1] == ret.branches[ibr].λ[:, iband, 1] ./ 100
-        end
+        @test ret.λ[:, iband, 1] == [10ik .+ iband for ik in 1:8]
+        @test ret.λerror[:, iband, 1] == ret.λ[:, iband, 1] ./ 100
     end
 
     B = model.recip_lattice
-    ref_kdist = zeros(3, 3)  # row idx is k-point, col idx is branch,
-    ikpt = 1
-    for ibr in 1:3
-        ibr != 1 && (ref_kdist[1, ibr] = ref_kdist[end, ibr-1])
-        ikpt += 1
-        for ik in 2:3
-            ref_kdist[ik, ibr] = (
-                ref_kdist[ik-1, ibr] + norm(B * (kcoords[ikpt-1] - kcoords[ikpt]))
-            )
-            ikpt += 1
+    ref_kdist = [0.0]
+    for ik in 2:8
+        if ik != 6
+            push!(ref_kdist, ref_kdist[end] + norm(B * (kcoords[ik-1] - kcoords[ik])))
+        else
+            # At ik = 6, the branch changes so kdistance does not increase.
+            push!(ref_kdist, ref_kdist[end])
         end
     end
-    for ibr in 1:3
-        @test ret.branches[ibr].kdistances == ref_kdist[:, ibr]
-    end
-
+    @test ret.kdistances == ref_kdist
     @test ret.ticks.labels == ["Γ", "X", "W | U", "X"]
-    @test ret.ticks.distances == [0.0, ref_kdist[end, 1], ref_kdist[end, 2], ref_kdist[end, 3]]
+    @test ret.ticks.distances == ref_kdist[[1,3,5,8]]
```

Comment:
Spacing after commas:
```suggestion
    @test ret.ticks.distances == ref_kdist[[1, 3, 5, 8]]
```

Reply:
Done